ui/raidboss/data/06-ew/raid/p3n.ts

```diff
@@ -0,0 +1,224 @@
+        return output.avoidCharge!();
+      },
+      outputStrings: {
+        avoidCharge: {
+          en: 'Avoid Charge',
+          de: 'Charge ausweichen',
+          fr: 'Évitez les charges',
+          ja: '突進避けて',
+          cn: '躲避冲锋',
+          ko: '돌진을 피하세요',
+        },
+      },
+    },
+    {
+      id: 'P3N Sunbird Spawn',
+      type: 'AddedCombatant',
+      netRegex: NetRegexes.addedCombatant({ name: 'Sunbird', capture: false }),
+      suppressSeconds: 1,
+      response: Responses.killAdds(),
+    },
+    {
+      id: 'P3N Fore/Rear Carve',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66A[78]', source: 'Sunbird', capture: false }),
+      suppressSeconds: 1,
+      alertText: (_data, _matches, output) => {
+        return output.text!();
+      },
+      outputStrings: {
+        text: {
+          en: 'Dodge Add Cleave',
+        },
+      },
+    },
+    {
+      id: 'P3N Dead Rebirth',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66A9', source: 'Phoinix', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P3N Ashen Eye',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66AB', source: 'Sparkfledged' }),
+      alertText: (data, matches, output) => {
+        if (!data.ashenEyeDirections)
+          data.ashenEyeDirections = [];
+        switch (matches.heading) {
+          case '-3.14':
+            data.ashenEyeDirections.push('north');
+            break;
+          case '-1.57':
+            data.ashenEyeDirections.push('west');
+            break;
+          case '0.00':
+            data.ashenEyeDirections.push('south');
+            break;
+          case '1.57':
+            data.ashenEyeDirections.push('east');
+            break;
+          default:
+            console.error('Ashen Eye: Got unforeseen ashen eye direction', matches.heading);
+            break;
+        }
+        if (data.ashenEyeDirections.length === 2) {
+          let safeSpot = '';
+          let first = '';
+          const dir1 = data.ashenEyeDirections[0];
+          const dir2 = data.ashenEyeDirections[1];
+          switch (dir1) {
+            case 'north':
+              safeSpot = output.s!();
+              break;
+            case 'west':
+              safeSpot = output.e!();
+              break;
+            case 'south':
+              safeSpot = output.n!();
+              break;
+            case 'east':
+              safeSpot = output.w!();
+              break;
+          }
+          switch (dir2) {
+            case 'north':
+              first = output.s!();
+              break;
+            case 'west':
+              first = output.e!();
+              break;
+            case 'south':
+              first = output.n!();
+              break;
+            case 'east':
+              first = output.w!();
+              break;
+          }
+          return first + ' then ' + safeSpot;
```

Comment:
Similar comment on P2N about using a `combo` style outputstring here.

I'd also shorten this to `X => Y` instead of using `then` as well.  TTS will transform that into "then" automatically, but it's easier to read (imo) on screen.

Reply:
Fixed in e32226eea0497fd9f12913a9d7136930d24c0a28